src/main/java/de/tum/cit/aet/artemis/communication/repository/ReactionRepository.java

```diff
@@ -20,4 +22,7 @@
 public interface ReactionRepository extends ArtemisJpaRepository<Reaction, Long> {
 
     List<Reaction> findReactionsByUserId(long userId);
+
+    @Query("SELECT r FROM Reaction r WHERE r.post.id IN :postIds")
+    List<Reaction> findByPostIds(@Param("postIds") List<Long> postIds);
```

Comment:
Here I also believe a Spring Data derived JPQL query could be used instead of manually specifiying the query:

```suggestion
    List<Reaction> findByPostIdIn(Collection<Long> postIds);
```